src/test/java/it/gov/pagopa/reward/service/reward/RewardContextHolderServiceImplTest.java

```diff
@@ -26,27 +29,32 @@ class RewardContextHolderServiceImplTest {
     @Mock private KieContainerBuilderService kieContainerBuilderServiceMock;
     @Mock private DroolsRuleRepository droolsRuleRepositoryMock;
     @Mock private ApplicationEventPublisher applicationEventPublisherMock;
+    @Mock private ReactiveRedisTemplate<String, byte[]> reactiveRedisTemplateMock;
+    @Value("${spring.redis.enabled}") private boolean isRedisCacheEnabled;
 
     private RewardContextHolderService rewardContextHolderService;
 
-    private final KieContainer expectedKieContainer = Mockito.mock(KieContainer.class);
+    private final KieBase expectedKieBase = Mockito.mock(KieBase.class);
 
     @BeforeEach
     void init(){
         Mockito.when(droolsRuleRepositoryMock.findAll()).thenReturn(Flux.empty());
-        Mockito.when(kieContainerBuilderServiceMock.build(Mockito.any())).thenReturn(Mono.just(expectedKieContainer));
+        Mockito.when(kieContainerBuilderServiceMock.build(Mockito.any())).thenReturn(Mono.just(expectedKieBase));
+        if (isRedisCacheEnabled) {
+            Mockito.when(reactiveRedisTemplateMock.opsForValue().get(Mockito.anyString())).thenReturn(Mono.just(SerializationUtils.serialize(expectedKieBase)));
+        }
```

Comment:
never executed, because isRedisCacheEnabled is always false

it could be useful to write another test, same as this, but using true instead